### ISSUE TITLE
Can't step into external code files while debugging

### ISSUE THREAD

**Issue by scruel** [microsoft/vscode-python-debugger#362]
```json
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python Debugger: Current File",
            "type": "debugpy",
            "request": "launch",
            "program": "${file}",
            "console": "integratedTerminal",
            "justMyCode": false,
        }
    ]
}
```

Logs:
```
2024-06-10 00:57:09.404 [info] Send text to terminal: 
2024-06-10 00:57:09.605 [info] Send text to terminal:  cd /home/scruel/Code/Python/xtorch ; /usr/bin/env /home/scruel/mambaforge/envs/xtorch/bin/python /home/scruel/.vscode-server/extensions/ms-python.debugpy-2024.6.0/bundled/libs/debugpy/adapter/../../debugpy/launcher 56247 -- /home/scruel/Code/Python/xtorch/main.py
2024-06-10 00:57:09.154 [info] DAP Server launched with command: /home/scruel/mambaforge/envs/xtorch/bin/python /home/scruel/.vscode-server/extensions/ms-python.debugpy-2024.6.0/bundled/libs/debugpy/adapter
```
![image](https://github.com/microsoft/vscode-python-debugger/assets/16933298/0c9b15b2-a9db-4b5e-8071-815bffce6319)

I can manually click to call stack to jump, and the active code line is green rather than yellow:
![image](https://github.com/microsoft/vscode-python-debugger/assets/16933298/cd492ec4-34cb-42bf-a546-4b3401a36b31)

**Comment by Clark-G** [comment]
I have the exactly same issue, hope to fix this quickly

**Comment by yasserben** [comment]
I am encountering the same issue as well

**Comment by avishek-mondal** [comment]
Hi, I am encountering the same issue as well.

This is my minimum reproducible example: 

```python
from transformers import BertTokenizerFast, BertModel


def main():
    model_name = "google-bert/bert-base-uncased"
    tokenizer = BertTokenizerFast.from_pretrained(model_name)
    model = BertModel.from_pretrained(model_name)

    words = [
        ["word1*word2|word3"],
        ["word1*", "word2|", "word3"],
    ]

    tokenizer_out = tokenizer(
        text=words,
        is_split_into_words=True,
        return_offsets_mapping=True,
        return_overflowing_tokens=True,
        truncation="longest_first",
        padding="max_length",
        return_tensors="pt",
        max_length=512,
        stride=0,
        return_length=True,
    )
    # Input IDs and attention mask
    input_ids = tokenizer_out["input_ids"]
    attention_mask = tokenizer_out["attention_mask"]

    # Inference with the mask
    model.eval()
    outputs = model(input_ids, attention_mask=attention_mask)

    print(outputs)


if __name__ == "__main__":
    main()

```

<img width="533" alt="image" src="https://github.com/microsoft/vscode-python-debugger/assets/17044043/d960f913-2b31-4690-99c4-daea52a0d96e">

and every time I try and step into the code from the `transformers` library, I get brought back to my own `main.py` module. I have to click the `Show 2 More Stack Frames` under the `Call Stack`, and then go back into the code in the `transformers` library.

**Comment by Ujifman** [comment]
Got the same issue after update VSCode to 1.90.0. 
Workaround: downgrade VSCode to 1.89.1
Here is full VSCode about screen of version where step in other libraries is working
```
Version: 1.89.1 (system setup)
Commit: dc96b837cf6bb4af9cd736aa3af08cf8279f7685
Date: 2024-05-07T05:13:33.891Z
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Windows_NT x64 10.0.22631
```

**Comment by BriceBoy** [comment]
Same problem here

VS Code Version 1.90.0
Debugpy Version v2024.6.0
Python Version 3.12.2
OS Windows

**Comment by dyitzchaki-roku** [comment]
downgrading just the extension doesn't help

**Comment by d-biehl** [comment]
very good! Thanks for the info!

**Comment by Andrej730** [comment]
Is there way to safely downgrade to VS Code 1.89.1 besides just uninstalling and installing it? Is there any chance to lose some settings/data during reinstall?

**Comment by avishek-mondal** [comment]
> Is there way to safely downgrade to VS Code 1.89.1 besides just uninstalling and installing it? Is there any chance to lose some settings/data during reinstall?

I don't think there is :( I kept a copy of `~/Library/Application\ Support/Code/User` and `~/.vscode/extensions` (I use a Mac) before I deleted VSCode 1.9 and installed 1.89

**Comment by avi-nextvestment** [comment]
I am running into the same issue:

```
VS Code Version 1.90.0
Debugpy Version v2024.6.0
Python extension v2024.8.0
Python Version 3.10.11
OS Windows
```

**Comment by avi-nextvestment** [comment]
Found this is a duplicate or https://github.com/microsoft/debugpy/issues/1598 and ultimately a VS Code issue that's documented here: https://github.com/microsoft/vscode/issues/214433

**Comment by paulacamargo25** [comment]
Closed in: [#214433](https://github.com/microsoft/vscode/issues/214433')

**Comment by Andrej730** [comment]
I'm on insiders version, can confirm that it now seems to work as before.

```python
Version: 1.91.0-insider (user setup)
Commit: 0a417782b1781d39803b8f1791cfb22f8835ddae
Date: 2024-06-13T05:48:47.964Z
Electron: 29.4.0
ElectronBuildId: 9593362
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.22635
```
